includes/admin/class-wc-rest-payments-accounts-controller.php

```diff
@@ -35,11 +35,38 @@ public function register_routes() {
 	}
 
 	/**
-	 * Get accounts details via API.
+	 * Get account details via API.
 	 *
-	 * @param WP_REST_Request $request Full data about the request.
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return WP_REST_Response|WP_Error
 	 */
 	public function get_account_data( $request ) {
-		return WC_Payments::get_account_service()->get_cached_account_data();
+		$account = WC_Payments::get_account_service()->get_cached_account_data();
+		if ( [] === $account && ! WC_Payments_Account::is_on_boarding_disabled() ) {
```

Comment:
```suggestion
		if ( [] === $account ) {
```

Reply:
Fixed in b0997bdc6c8959ec7be0788608868e0626faf5e9